models_validators/validators.py

```diff
@@ -0,0 +1,47 @@
+import re
+from pydantic import BaseModel, EmailStr
+from fastapi import HTTPException
+from datetime import date, datetime
+
+
+class UserValidation(BaseModel):
+    username: str
+    email: EmailStr
+    password: str
+    name: str
+    firstname: str
+    date_of_birth: date
+    phone_number: str
+    address: str
+
+    def check_valid_pwd(self):
+        pwd = self.password
+        if len(pwd) < 8:
+            raise HTTPException(
+                status_code=403,
+                detail="Password must contain at least 8 characters."
+            )
```

Comment:
Plusieurs problèmes avec cette approche:
1. L'exception `HTTPException` est une exception spécialisée qui est de la responsabilité de l'API pas la tienne.
2. Tu masques la vraie cause de l'échec de la validation en mettant tout dans le commentaire de l'exception et en forçant le code de statut à 403. La bonne approche consiste à créer des validations spécialisées et transmettre les exceptions appropriées selon la validation qui échoue. L'API va insérer le bon message dans son payload.